### DISCUSSION TITLE
How would one set up a global interpreter with PDM that has preinstalled libraries?

### DISCUSSION THREAD

**Discussion by robml** [pdm-project/pdm#977] [Q&A] [answered]
So I have recently been developing and noticed I use a few packages on a regular basis. They don't warrant a complete installation of `conda` or Anaconda (which I think may be overkill) I know usually with `venv`s you would simply activate the virtual environment and that would serve as your interpreter with say `numpy` or `pandas` already installed if I wanted them to be. Is there something similar with `pdm` and if so how would we implement it? 

The above is just to play around with the packages regardless of my directory without having to initialize a project. On a separate note for dedicated projects I don't mind running `pdm init`, my only question is if how PDM works in this sense. Does it completely reinstall all the packages from scratch (ie I have two projects A and B both with a version of `django` both running `v3.0.0`) so that I end up with two duplicates, 

Or is it the case that if I initialize project B and it detects I already have `django v3.0.0` installed it will just create a link to that package instead (until I upgrade one or the other breaking the similarity, i.e. one is `v4.0.0` and one is `v3.0.0`)? If there are duplicates how does PDM (or any other package manager) address this redundancy and how much disk space waste am I looking at? Gigabytes?

**Comment by frostming** [comment, marked as answer]
The first question, you just need a [global project](https://pdm.fming.dev/usage/project/#manage-global-project)
For the second, here you are [the install caches](https://pdm.fming.dev/usage/project/#cache-the-installation-of-wheels)

**Reply by robml** [reply]
I see, I recently tried the `-g` option however it failed due to requiring `sudo` permissions. Trouble is because I have `pdm` installed on my user account my shell doesn't recognize `sudo pdm` as a valid command despite my user being part of the `sudo` group. Would you recommend installing `pdm` separately as the `root`/`su` user or is there another workaround to this? I have tried quite hard to keep my system clean and want to minimize anything installed on the root's profile that isn't necessary.

EDIT: I got around it by including my user path in the sudo call, seems to work now!
